lib/wpc_proto/config.h

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2019 Wirepas Ltd licensed under Apache License, Version 2.0
+ *
+ * See file LICENSE for full license details.
+ *
+ */
+
+#ifndef SOURCE_CONFIG_H_
+#define SOURCE_CONFIG_H_
+
+#include "wpc.h"
+#include "wp_global.pb.h"
+#include "config_message.pb.h"
+
+/* Return the size of a struct member */
+#define member_size(type, member) (sizeof( ((type *)0)->member ))
+
+/**
+ * \brief   Return network address state
+ * \return  true if network address is set
+ */
+bool Config_Get_has_network_address();
+
+/**
+ * \brief   Return network address in config cache
+ * \return  network address, valid only if stack status indicate that network address is set
+ */
+net_addr_t Config_Get_network_address();
+
+/**
+ * \brief   Handle config cache when status changed
+ */
+void Config_On_stack_boot_status(uint8_t status);
+
+/**
+ * \brief   Fill event message header
+ * \param   header_p pointer to header to fill
+ */
+void Config_Fill_event_header(wp_EventHeader * header_p);
+
+/**
+ * \brief   Fill reponse message header
+ * \param   header_p pointer to header to fill
+ * \param   req_id request id
+ * \param   res response to request
+ */
+void Config_Fill_response_header(wp_ResponseHeader * header_p,
```

Comment:
Does it belong to config.h?

Shouldn't it be in a more generic module that could be used by other kind of messages, like data response?

Reply:
this one (Config_Fill_response_header ) is used by both handle_send_data_request() and Config_Handle_set_config_request(). So it's generic. To be discuss where to put it.